internal/analytics/client.go

```diff
@@ -28,8 +28,8 @@ type VersionInfo struct {
 	CommitId  string `json:"commit_id,omitempty"`
 }
 
-// currentHub is the initial Hub with no Client bound and an empty Scope.
-var currentHub = New()
+// Consumers must call analytics.Init before using this package
+var currentHub *Client = nil
```

Comment:
Not sure this is the right fix.
We could also call `Init()` before each method call in this package if `currentHub = nil` (and maybe add a lock to make it safe)

Reply:
Ye that is fine, you can actually do in initAnalytics in cmd package to set the default hub or even wrap it in a sync.Once

Reply:
```suggestion
var currentHub *Client
```

Reply:
I'm probably missing something here. We already call `initAnalytics` here https://github.com/cloudquery/cloudquery/blob/811855475622955dcfb323298292bde958f4372d/cmd/root.go#L157.

Where should I add another call?

Reply:
I think this is fine